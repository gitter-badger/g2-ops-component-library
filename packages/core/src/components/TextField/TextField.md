Generic TextField component that can be used to implement custom components such as Currency, Phone, etc.

Example:
```js
initialState = { textFieldValue: '' };
<div style={{ width: '300px' }}>
  <TextField
    horizontal
    label='First Name'
    placeholder='Enter First Name'
  />
  <TextField
    horizontal
    label='Last Name'
    placeholder='Enter Last Name'
  />
  <TextField
    horizontal
    label='Required Field'
    placeholder='This Is Required'
    onGetErrorMessage={() => (state.textFieldValue || ''.length) === 0 ? 'This field is required.' : ''}
  />
  <br/>
    <TextField
      label='First Name'
      placeholder='Enter First Name'
    />
    <TextField
      label='Last Name'
      placeholder='Enter Last Name'
    />
    <TextField
      label='Required Field'
      value={state.textFieldValue}
      onChanged={(textFieldValue) => setState({ textFieldValue })}
      onGetErrorMessage={() => (state.textFieldValue || ''.length) === 0 ? 'This field is required.' : ''}
    />
</div>
```
