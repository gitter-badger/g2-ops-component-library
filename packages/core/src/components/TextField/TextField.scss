input {
	color: black !important;
	padding: 0 12px 0 5px !important;
	font-family: 'Roboto', sans-serif;
	font-size: 16px;
}

.ms-TextField.leftLabel {
	margin-bottom: 8px;
}

.ms-TextField.leftLabel .ms-TextField-wrapper {
	display: flex;
	align-items: center;
	justify-content: space-between;
}

.ms-TextField-errorMessage {
	font-family: 'Roboto', sans-serif;
}

.ms-TextField-suffix {
	background: #ffffff !important;
	color: #000000 !important;
	align-items: center;
}

.ms-TextField .ms-Label {
	font-family: 'Roboto', sans-serif;
	font-size: 12px;
	color: #1d5ab9;
	white-space: nowrap;
}