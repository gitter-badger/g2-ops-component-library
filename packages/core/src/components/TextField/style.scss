:global {
	input {
		color: black !important;
		padding: 0 12px 0 5px !important;
		font-family: 'Roboto', sans-serif;
		font-size: 16px;
	}
	
	.ms-TextField {
		label {
			font-family: 'Roboto', sans-serif;
			font-size: 12px;
		}
	}
	.ms-TextField-errorMessage {
		font-family: 'Roboto', sans-serif;
	}
	.ms-TextField-suffix {
		background: #ffffff !important;
		color: #000000 !important;
		align-items: center;
	}
}
