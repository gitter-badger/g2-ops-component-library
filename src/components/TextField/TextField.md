Generic TextField component that can be used to implement custom components such as Currency, Phone, Address, etc.

Example:

```js
<TextField placeholder={'This is a text field'} />
```