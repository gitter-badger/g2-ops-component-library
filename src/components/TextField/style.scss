input {
  color: black !important;
  padding: 0 12px 0 5px !important;
}
