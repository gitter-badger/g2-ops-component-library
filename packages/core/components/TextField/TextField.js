// @flow

import React from 'react'
import { TextField as FabricTextField } from 'office-ui-fabric-react/lib/TextField'
import { Fabric } from 'office-ui-fabric-react/lib/Fabric'

import wrapFabricContext from '../../wrapFabricContext'

import './style.scss'

type TextFieldPropTypes = {
  text: string,
  disabled: boolean,
  placeholder?: string,
  className?: string,
  inputClassName: string,
  otherProps?: {
    [string]: any,
  },
}

const TextField = wrapFabricContext(
  ({ placeholder, disabled, className, inputClassName, ...otherProps }: TextFieldPropTypes) => (
    <FabricTextField
      {...otherProps}
      disabled={disabled}
      inputClassName={`input ${inputClassName}`}
      className={className}
      placeholder={disabled ? '' : placeholder}
    />
  ),
)

TextField.defaultProps = {
  inputClassName: '',
}

export default TextField
