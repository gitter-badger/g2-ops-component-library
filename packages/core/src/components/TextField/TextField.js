// @flow

import React from 'react'
import { TextField as FabricTextField } from 'office-ui-fabric-react/lib/TextField'
import { Fabric } from 'office-ui-fabric-react/lib/Fabric'

import {wrapFabricContext} from 'utilities/wrapFabricContext'

import './TextField.scss'

type TextFieldPropTypes = {
	text: string,
	label: string,
  disabled: boolean,
  placeholder?: string,
  className?: string,
  inputClassName: string,
  otherProps?: {
    [string]: any,
  },
}

export const TextField = wrapFabricContext(
  ({ placeholder, disabled, className, inputClassName, horizontal, ...otherProps }: TextFieldPropTypes) => (
    <FabricTextField
      {...otherProps}
      disabled={disabled}
      inputClassName={`input ${inputClassName}`}
      className={`${className} ${horizontal ? 'leftLabel' : ''}`}
      placeholder={disabled ? '' : placeholder}
    />
  ),
)

TextField.defaultProps = {
  inputClassName: '',
}
