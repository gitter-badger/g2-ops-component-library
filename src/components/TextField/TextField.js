import React from 'react'
import PropTypes from 'prop-types'
import { TextField as FabricTextField } from 'office-ui-fabric-react/lib/TextField';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric'

const propTypes = {
  text: PropTypes.string,
}

const TextField = (props) => (
  <Fabric>
    <FabricTextField {...props} />
  </Fabric>
)

export default TextField