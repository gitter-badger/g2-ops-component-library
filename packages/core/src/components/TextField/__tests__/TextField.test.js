import React from 'react'
import renderer from 'react-test-renderer'

import {TextField} from '../TextField'

// This test case is just a placeholder nothing adding much. As we are not doing anything in it. We are adding wrapper around Fabric TextField
describe('<TextField />', () => {
  test('<TextField /> should render properly', () => {
    const tree = renderer.create(<TextField />)
    expect(tree).toMatchSnapshot()
  })
})
