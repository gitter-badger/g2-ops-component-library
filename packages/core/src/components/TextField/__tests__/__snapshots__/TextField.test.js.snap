// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> <TextField /> should render properly 1`] = `
<div
  className="ms-Fabric root-33"
>
  <div
    className="ms-TextField root undefined "
  >
    <div
      className="ms-TextField-wrapper wrapper"
    >
      <div
        className="ms-TextField-fieldGroup fieldGroup"
      >
        <input
          aria-invalid={false}
          className="ms-TextField-field field input "
          id="TextField0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
